hatchet_sdk/workflow_run.py

```diff
@@ -32,16 +32,18 @@ def result(self) -> Coroutine:
         return self.workflow_listener.result(self.workflow_run_id)
 
     def sync_result(self) -> dict:
+        coro = self.workflow_listener.result(self.workflow_run_id)
         loop = get_active_event_loop()
+
         if loop is None:
-            with EventLoopThread() as loop:
-                coro = self.workflow_listener.result(self.workflow_run_id)
-                future = asyncio.run_coroutine_threadsafe(coro, loop)
-                return future.result()
+            loop = asyncio.new_event_loop()
+            asyncio.set_event_loop(loop)
+            try:
+                return loop.run_until_complete(coro)
+            finally:
+                asyncio.set_event_loop(None)
         else:
-            coro = self.workflow_listener.result(self.workflow_run_id)
-            future = asyncio.run_coroutine_threadsafe(coro, loop)
-            return future.result()
+            return loop.run_until_complete(coro)
```

Comment:
I'm not 100% sure this is safe, but I think the only risk of it causing issues is if there's no event loop, which I can't imagine being super common. It's a little tough to test well though